homeassistant/components/home_connect/__init__.py

```diff
@@ -50,15 +50,19 @@
 SERVICE_SETTING_SCHEMA = vol.Schema(
     {
         vol.Required(ATTR_DEVICE_ID): str,
-        vol.Required(ATTR_KEY): str,
+        vol.Required(ATTR_KEY): vol.In(
+            [setting.value for setting in SettingKey if setting != SettingKey.UNKNOWN]
+        ),
```

Comment:
I'd try using `vol.Coerce(SettingKey)` and then another validator that disallows `SettingKey.UNKNOWN`.

```suggestion
        vol.Required(ATTR_KEY): vol.All(
            vol.Coerce(SettingKey),
            vol.NotIn([SettingKey.UNKNOWN]),
        ),
```